add paper figure folder and notebook for figure 3 tasmasx seasonal trend with updates

emileten opened the pull request:
- [ ] closes #xxxx
 - [ ] tests added / passed
 - [ ] docs reflect changes
 - [ ] passes ``flake8 downscale tests docs``
 - [ ] entry in HISTORY.rst

- added a `notebooks/paper_figures` folder to centralize code producing figures for the GDPCIR paper. 
- in this folder, added a notebook named `tasmax_seasonal_quantile_trend` that produces the figure currently labelled 'figure 3', visualizing seasonal quantile trends for tasmax comparing model to downscaled with reanalysis levels side by side.

kemccusker commented on notebooks/paper_figures/tasmax_seasonal_quantile_trend.ipynb:
```diff
@@ -0,0 +1,394 @@
+    "#### last updated : 2022/11/15, by Emile Tenezakis (e.tenezakis@gmail.com). \n",
+    "#### scaling : with a remote dask cluster as a backend to the xarray datasets. With the cluster scaling parameters as is, notebook should take around 10 minutes, including normal cluster spin up time. \n",
+    "#### output : notebook saves the figure to user specific `figure_3_output_file_path` defined below.\n",
+    "#### library dependencies : This ran on rhodium's onyx environment, with the rhodium-internal `catalyst_compute_tools` and `dodola` packages pip-installed in editable mode (install `dodola` without its dependencies with the `--no-deps` flag).  \n",
```
@emileten is there a way to produce the figures w/o the rhodium-internal packages (assuming they are not public)?

emileten replied:
@kemccusker actually, the only private package this relies on is catalyst_compute_tools https://gitlab.com/crs-catalyst/infrastructure/catalyst_compute_tools. I use to spin up the cluster.

Also, I'll have to change that notebook anyways given the result of our conversation yesterday. So we can pause this review until that's done and I can also switch tools and use [this](https://rhg-compute-tools.readthedocs.io/en/latest/rhg_compute_tools.html#module-rhg_compute_tools.kubernetes) instead that I think is public.

kemccusker reviewed (approved):
Looks good! I just left one question about the rhodium-internal packages. Seems like it would be useful to not make the figure production dependent on those but that could be a later PR

emileten commented:
@kemccusker I addressed your above comment (using our open source rather than closed source compute package), but I will work on the precipitation figure (currently numbered figure 4) in the same PR if that's ok to keep things in the same place, so not merging this yet.
